operator/controllers/csiscaleoperator_controller.go

```diff
@@ -219,6 +221,62 @@ func (r *CSIScaleOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	logger.Info("Creation of the resources which never change is successful")
 
+	// Rollout restart of node plugin pods, if modified driver
+	// manifest file is applied.
+	// The CSIConfigMap has the data which is last applied, so
+	// compare the clusters data of current driver instance with
+	// CSIConfigMap data and rollout restart if there is a difference.
+	configMap := &corev1.ConfigMap{}
+	cerr := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      config.CSIConfigMap,
+		Namespace: req.Namespace,
+	}, configMap)
+	if cerr != nil {
+		if !errors.IsNotFound(cerr) {
+			message := "Failed to get ConfigMap resource " + config.CSIConfigMap
+			logger.Error(err, message)
+			// TODO: Add event.
+			meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
+				Type:    string(config.StatusConditionSuccess),
+				Status:  metav1.ConditionFalse,
+				Reason:  string(csiv1.ResourceReadError),
+				Message: message,
+			})
+			logger.Error(err, message)
+			return ctrl.Result{}, cerr
+		}
+	} else {
+		clustersBytes, err := json.Marshal(&instance.Spec.Clusters)
+		if err != nil {
+			logger.Error(err, "Failed to marshal clusters data of this instance")
+			return ctrl.Result{}, err
+		}
+		clustersString := string(clustersBytes)
+		configMapDataBytes, err := json.Marshal(&configMap.Data)
+		if err != nil {
+			logger.Error(err, "Failed to marshal data of ConfigMap "+config.CSIConfigMap)
+			return ctrl.Result{}, err
+		}
+		configMapDataString := string(configMapDataBytes)
+		configMapDataString = strings.Replace(configMapDataString, " ", "", -1)
+		configMapDataString = strings.Replace(configMapDataString, "\\\"", "\"", -1)
+
+		if !strings.Contains(configMapDataString, clustersString) {
+			logger.Info("CSIScaleOperator instance is changed, so restarting node plugin pods")
+			daemonSet, derr := r.getNodeDaemonSet(instance)
+			if derr != nil {
+				logger.Error(derr, "Failed to get node plugin DaemonSet")
+			} else {
+				err = r.rolloutRestartNode(daemonSet)
+				if err != nil {
+					logger.Error(err, "Failed to rollout restart of node plugin pods")
+				} else {
+					daemonSetRestartedKey, daemonSetRestartedValue = r.getRestartedAtAnnotation(daemonSet.Spec.Template.ObjectMeta.Annotations)
```

Comment:
`daemonSetRestartedKey`, `daemonSetRestartedValue` are global variables, these will get initialized back to empty string when operator pod is restarted, this would result in pods to restart again. Can you please confirm if this is happening here or not?

Reply:
These are code changes to restart CSI driver pods when a modified CR is applied. And I have tested and pods are getting restarted as expected. Though checking the behavior when restart of operator pod is interesting, it is not related to these changes, so we can check this separately.

Reply:
I don't think this issue is tracked somewhere, that's why I wanted you to once test this with your changes. If it occurs, we can add a TODO comment in this PR or add it our list of enhancements.

Reply:
Though this is a separate thing, I tried restart of operator pod and other pods don't get restarted due to operator restart, which is good.